{{cookiecutter.repostory_name}}/bin/backup-db.sh

```diff
@@ -11,10 +11,9 @@ fi
 . .env
 
 target="db_dump_$(date +%Y-%m-%d_%H%M%S).sql.gz"
-if [ -n "$DATABASE_URL" ]; then
-  docker run --rm postgres:9.6 pg_dump -d "$DATABASE_URL" | gzip > "$target"
+
+if [[ "$DATABASE_URL" =~ "@db:" ]]; then
+  docker run --rm --network {{cookiecutter.repostory_name}}_default postgres:9.6-alpine pg_dump "$DATABASE_URL" | gzip > "$target"
 else
-  docker-compose exec db pg_dumpall -c -U postgres | gzip > "$target"
+  docker run --rm --network host postgres:9.6-alpine pg_dump "$DATABASE_URL" | gzip > "$target"
```

Comment:
I don't get the main idea I think. I see you used
* `docker-compose` + `pg_dumpall` if there was no `DATABASE_URL`, and
* `docker` + `pg_dump` if having `DATABASE_URL`.

What are the cases when you _don't want_ to `docker-compose exec pg_dumpall`?

Reply:
First of all, this script was broken. It probably worked in some old implementation of our template, but not with the latest one.

So now, `DATABASE_URL` is always defined, but it can:
1. use `db` host when we use postgres container and the 5432 port is not exposed (default production setup)
1. use `localhost` (default development setup)
1. use cloud postgres DB instance

In all cases, we want to backup only the DB defined in `DATABASE_URL`, so that why I removed usage of `pg_dumpall`. We usually not use more than one DB.

As for `docker-compose`, I could have used it, but I don't want to rely on the docker-compose configuration, and I would have to use different logic for 2 and 3. Now we're kind of independent from the app configuration.

Reply:
I think we store tenants in separate databases in MRC, so without pg_dumpall and an equivalent solution for a remote database, we are going to only store the table which says which tenant is stored in which database and none of the actual user data

Reply:
This script supports only what the template supports, and you can not generate a project without `DATABASE_URL` like in MRC. The MRC is so customized, way beyond this template. It even uses a different `DB_` prefix instead of `POSTGRES_` as we use here.

When we add support for multiple databases to the template, we should update this script, but now you can not just remove `DATABASE_URL` without further customization.